File: Selenium/Activities/Activity13.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Open the browser to the URL
    driver.get("https://v1.training-support.net/selenium/tables")
    print("Title of the page is: ", driver.title)
    # Find the number of rows and columns in the table
    rows = driver.find_elements(By.XPATH, "//table[contains(@class,'striped')]/tbody/tr")
    print("No.of rows: ", len(rows))
    cols = driver.find_elements(By.XPATH, "//table[contains(@class,'striped')]/tbody/tr[1]/td")
    print("No.of cols: ", len(cols))
    # Find and print all the cell values in the third row of the table
    thirdRow = driver.find_elements(By.XPATH, "//table[contains(@class,'striped')]/tbody/tr[3]/td")
    print("Cell values in the third row of the table are: ")
    for cell in thirdRow:
        print(cell.text)
    # Find and print the cell value at the second row second column
    secRowSecCol = driver.find_element(By.XPATH, "//table[contains(@class,'striped')]/tbody/tr[2]/td[2]")
    print("Cell value at the second row second column: ", secRowSecCol.text)

File: Selenium/Activities/Activity13.java
package Activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class Activity13 {
    public static void main(String[] args) throws InterruptedException {
        // Setup the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/tables");
        String Title = driver.getTitle();
        System.out.println("Title of the page: " +Title);
        // Find the number of rows and columns in the first table
        List<WebElement> rows = driver.findElements(By.xpath("//table[contains(@class,'striped')]/tbody/tr"));
        System.out.println("No.of rows: " +rows.size());
        List<WebElement> cols = driver.findElements(By.xpath("//table[contains(@class,'striped')]/tbody/tr[1]/td"));
        System.out.println("No.of columns: " +cols.size());
        // Find and print all the cell values in the third row of the table
        List<WebElement> thirdRow = driver.findElements(By.xpath("//table[contains(@class,'striped')]/tbody/tr[3]/td"));
        System.out.println("Cell values in the third row of the table are: ");
        for(WebElement cell: thirdRow){
            System.out.println(cell.getText());
        }
        // Find and print the cell value at the second row second column
        WebElement secRowSecCol = driver.findElement(By.xpath("//table[contains(@class,'striped')]/tbody/tr[2]/td[2]"));
        System.out.println("Cell value at the second row second column: " +secRowSecCol.getText());
        // Close the browser
        driver.quit();
    }
}
